docs/src/references/agent/software-management.md

```diff
@@ -0,0 +1,378 @@
+---
+title: Software management
+tags: [Reference, Agent, Software Management]
+sidebar_position: 3
+---
+
+Thin-edge software management is implemented by two operations
+which give the ability to manage software packages of different types on the same device.
+
+- `software_list` is used to fetch a pertinent subset of the software packages installed on a device.
+- `software_update` let the user install, update and remove software packages on a device.
+
+## software_list MQTT API
+
+The `software_list` operation API follows the [generic thin-edge rules for operations](device-management-api.md):
+- The `te/<device-topic-id>/cmd/software_list` topic is used to publish the type of software packages
+  that can be managed on the device with the given topic identifier.
+- Each `te/<device-topic-id>/cmd/software_list/+` topic is dedicated to a software list command instance,
+- The workflow is [generic with `"init"`, `"executing"`, `"successful"` and `"failed"` statuses](references/agent/device-management-api.md/#operation-workflow).
+
+### Operation registration
+
+The registration message of the `software_list` operation on a device:
+- must provide a `supportedTypes` list of the types of software package that can be installed on this device (e.g. `["apt", "docker"]`)
+- can provide a description of the operation and of each supported package type.
+
+```sh te2mqtt formats="v1"
+tedge mqtt pub --retain 'te/device/child001///cmd/software_list' '{
+    "description": "List software packages installed on the device",
+    "supportedTypes": [
+      "apt",
+      "docker"
+    ]
+}'
+```
+
+### init state
+
+A `software_list` command has nothing to provide beyond a `status` field.
+This empty message stands for a request of the list of software currently installed.
+
+```sh te2mqtt formats="v1"
+tedge mqtt pub --retain 'te/device/child001///cmd/software_list/c8y-2023-09-25T14:34:00' '{
```

Comment:
[Out of topic] I found using cloud type and RFC3339 timestamp as a command ID brilliant and better than random ID generated by nanoid. It gives understanable information to users and also unique!

Reply:
See https://github.com/thin-edge/thin-edge.io/issues/2251